### ISSUE TITLE
A suggestion for improvement

### ISSUE THREAD

**Issue by savehate** [comport9/HFS#1]
def select_talent(self, loc):
    x, y = TALENT_SELECT
    x, y = self.scale_XY(x, y)
    y += (round(self.scaleY * 77) * loc)
    self.mouse_move(x, y)
    self.mouse_click()

By moving the mouse click way to feel that it is easy to have problems, identify the location by entering a ctrl on the keyboard to determine the position after a few s after clicking through ctrl+x to click better.

**Comment by comport9** [comment]
If I'm understanding correctly, you're suggesting having the player hit "CTRL" first, then move the mouse automatically, then have the player hit "CTRL+X" to come back to their original mouse location?

I'm sure this would solve the problem of the mouse clicking in the wrong location occasionally, but unfortunately the game is moves to fast for that, and the whole point of the program is to make selecting the talent automatic. 

Thanks for your input though! I think you must be the first person to actually see this code. lol.

**Comment by savehate** [comment]
I think maybe I didn't express it well enough, but I wanted to mean that the "X" in "ctrl+X" represents the Arabic numerals 1, 2, 3, which is a combination of talents that can be quickly selected in Heroes of the Storm, such as pressing "ctrl" and the number key "1" at the same time to select the first talent counted from above, and pressing "ctrl" and "2" at the same time to select the second talent counted from above. As long as you identify the "heart.png" of the current talent layer in the judgment box of which combination of keys can be, this advantage is that the whole process does not require the participation of mouse buttons, to prevent the possibility of not finding the mouse suddenly in the fierce team battle.

    By the way, I hope that Microsoft will get better after the acquisition of Activision Blizzard! Thanks also to your friends for these efforts, your code can help some heroes with relatively fixed talents to operate faster!

**Comment by savehate** [comment]
I think maybe I didn't express it well enough, but I wanted to mean that the "X" in "ctrl+X" represents the Arabic numerals 1, 2, 3, which is a combination of talents that can be quickly selected in Heroes of the Storm, such as pressing "ctrl" and the number key "1" at the same time to select the first talent counted from above, and pressing "ctrl" and "2" at the same time to select the second talent counted from above. As long as you identify the "heart.png" of the current talent layer in the judgment box of which combination of keys can be, this advantage is that the whole process does not require the participation of mouse buttons, to prevent the possibility of not finding the mouse suddenly in the fierce team battle.
By the way, I hope that Microsoft will get better after the acquisition of Activision Blizzard! Thanks also to your friends for these efforts, your code can help some heroes with relatively fixed talents to operate faster!

**Comment by comport9** [comment]
Ahh yes, that does make more sense. However, I think I opted not to do that because the location of the heart icon doesn't always match the "number" for the talent because there aren't always 3 selections, sometimes there can be 4 or 5. The "ctrl-2" for example, is always the second from the top option, but my program has no way of knowing if there is any option "above", and so no way to know what number needs to be pressed.

I actually experimented with that very thing initially, but it required reading more images from the screen and more logic. Doable, but considering the very small downside of doing it the current way compared to the amount of the work required to do it the other, just didn't seem worth it to me.

Are you using the program, if so, how has it worked for you?
